### PR TITLE
Adds instance pool option to Databricks integration

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Adds option for users to specify a Databricks instance pool. All JobClusters that Aqueduct creates will create clusters using instances from this pool. If not specified, we default to a node type and request spot instances from AWS.
## Related issue number (if any)
Eng-2765
## Loom demo (if any)
https://www.loom.com/share/a12b0ca34fc04ad495b39b69422e72ec
## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/job/config.go [hunk 1]
```diff
@@ -64,6 +64,8 @@ type DatabricksJobManagerConfig struct {
 	// storage in S3. Information on how to create this can be found here:
 	// https://docs.databricks.com/aws/iam/instance-profile-tutorial.html
 	S3InstanceProfileARN string `yaml:"s3InstanceProfileArn" json:"s3_instance_profile_arn"`
+	// [Optional] ID of instance pool that Aqueduct-created JobClusters should use.
+	InstancePoolID *string `yaml:"instancePoolID" json:"instance_pool_id"`
```
Comment (saurav-c) [inline src/golang/lib/job/config.go:68]: Any specific reason for this to be a pointer?

File: src/ui/common/src/components/integrations/dialogs/databricksDialog.tsx [hunk 2]
```diff
@@ -85,6 +86,28 @@ export const DatabricksDialog: React.FC<Props> = ({
...
+        <Link href="https://docs.databricks.com/aws/iam/instance-profile-tutorial.html">
+          the Databricks documentation
+        </Link>
+        .
+      </Typography>
+
+      <IntegrationTextInputField
+        label={'Instance Pool ID'}
+        description={
+          'The ID of the Datbricks Instance Pool that Aqueduct will run compute on.'
```
Comment (saurav-c) [inline src/ui/common/src/components/integrations/dialogs/databricksDialog.tsx:101]: nit: typo for `Databricks`

**Review by saurav-c — APPROVED** [review]
Couple nits otherwise LGTM

File: src/golang/lib/job/config.go (hunk 1 above)
Comment (hsubbaraj-spiral) [reply src/golang/lib/job/config.go:68]: It's an optional value, I've read somewhere that using pointers for optional values is better than just dealing with empty strings.

**Pull request merged**